File: java/graphAlgorithms/misc/cgraph.java
// usage: cgraph type
//
// Copy a graph of specified type from stdin to stdout.
// Why you ask? To test input and output methods, of course.
//
// The allowed values of type are graph, wgraph,
// digraph, wdigraph, flograph, wflograph.

package algoLib.graphAlgorithms.misc;

import algoLib.misc.Util;
import algoLib.dataStructures.graphs.Graph;
import algoLib.dataStructures.graphs.Wgraph;

import java.io.*;

public class cgraph {
	public static void main(String [] args) {
		if (args.length != 1)
			Util.fatal("usage: cgraph type");

		PushbackReader in = new PushbackReader(
				    new InputStreamReader(System.in));
	
		if (args[0].equals("graph")) {
			Graph g = new Graph(1,1); g.read(in);
			System.out.println("" + g);
		} else if (args[0].equals("wgraph")) {
			Wgraph wg = new Wgraph(1,1); wg.read(in);
			System.out.println("" + wg);
/*
		} else if (args[0].equals("digraph")) {
			Digraph dg = new Digraph(1,1); dg.read(in);
			System.out.println("" + dg);
		} else if (args[0].equals("wdigraph")) {
			Digraph wdg = new Digraph(1,1); wdg.read(in);
			System.out.println("" + wdg);
		} else if (args[0].equals("flograph")) {
			Flograph fg = new Flograph(1,1); fg.read(in);
			System.out.println("" + fg);
		} else if (args[0].equals("wflograph")) {
			Flograph wfg = new Flograph(1,1); wfg.read(in);
			System.out.println("" + wfg);
*/
		} else {
			Util.fatal("usage: cgraph type");
		}
	}
}

File: cpp/graphAlgorithms/misc/cgraph.cpp
// usage: cgraph type
//
// Copy a graph of specified type from stdin to stdout.
// Why you ask? To test input and output operators, of course.
// We also do an assignment in between input and output,
// in order to test the assignment operator.
//
// The allowed values of type are graph, wgraph,
// digraph, wdigraph, flograph, wflograph.

#include "stdinc.h"
#include "Adt.h"
#include "Wgraph.h"
#include "Wdigraph.h"
#include "Wflograph.h"

using namespace grafalgo;

int main(int argc, char *argv[]) { 

	if (argc != 2) Util::fatal("usage: cgraph type");

	if (strcmp(argv[1],"graph") == 0) {
		Graph g; cin >>g; Graph g1;
		g1.copyFrom(g); cout << g1;
	} else if (strcmp(argv[1],"wgraph") == 0) {
		Wgraph wg; cin >> wg; Wgraph wg1;
		wg1.copyFrom(wg); cout << wg1;
	} else if (strcmp(argv[1],"digraph") == 0) {
		Digraph dig; cin >> dig; Digraph dig1;
		dig1.copyFrom(dig); cout << dig1;
	} else if (strcmp(argv[1],"wdigraph") == 0) {
		Wdigraph wdig; cin >> wdig; Wdigraph wdig1;
		wdig1.copyFrom(wdig); cout << wdig1;
	} else if (strcmp(argv[1],"flograph") == 0) {
		Flograph fg; cin >> fg; Flograph fg1;
		fg1.copyFrom(fg); cout << fg1.toString();
	} else if (strcmp(argv[1],"wflograph") == 0) {
		Wflograph wfg; cin >> wfg; Wflograph wfg1;
		wfg1.copyFrom(wfg); cout << wfg1;
	} else {
		Util::fatal("usage: cgraph type");
	}
}
